frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/PolicyAccessPackageAccordion/PolicyAccessPackageAccordion.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import classes from './PolicyAccessPackageAccordion.module.css';
+import type { PolicyAccessPackage } from '@altinn/policy-editor/types';
+import { PolicyAccordion } from '../PolicyAccordion/PolicyAccordion';
+import { useResourceAccessPackageServicesQuery } from 'app-shared/hooks/queries/useResourceAccessPackageServicesQuery';
+import { StudioSpinner } from '@studio/components';
+
+interface PolicyAccessPackageAccordionProps {
+  accessPackage: PolicyAccessPackage;
+  selectedLanguage: 'nb' | 'nn' | 'en';
+  selectPackageElement: React.ReactNode;
+}
+
+export const PolicyAccessPackageAccordion = ({
+  accessPackage,
+  selectedLanguage,
+  selectPackageElement,
+}: PolicyAccessPackageAccordionProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const [isServicesEnabled, setIsServicesEnabled] = useState<boolean>(false);
+
+  const { data: services, isLoading } = useResourceAccessPackageServicesQuery(
+    accessPackage.urn,
+    localStorage.getItem('policyEditorAccessPackageEnv') || 'prod', // hardcoded to prod for now
+    isServicesEnabled,
+  );
+
+  const onOpenAccordion = () => {
+    setIsServicesEnabled(true);
+  };
+
+  return (
+    <div className={classes.accessPackageAccordion}>
+      <PolicyAccordion
+        title={accessPackage.name}
+        subTitle={accessPackage.description}
+        extraHeaderContent={selectPackageElement}
+        onOpened={onOpenAccordion}
+      >
+        {isLoading && (
+          <StudioSpinner spinnerTitle={t('policy_editor.access_package_loading_services')} />
+        )}
+        {services?.length > 0 && (
+          <>
+            <div className={classes.serviceContainerHeader}>
+              {t('policy_editor.access_package_services')}
+            </div>
+            {services.map((resource) => {
+              return (
+                <div key={resource.identifier} className={classes.serviceContainer}>
+                  {resource.logoUrl ? (
+                    <img
+                      className={classes.logo}
+                      src={resource.logoUrl}
+                      alt={resource.hasCompetentAuthority.name[selectedLanguage]}
+                      title={resource.hasCompetentAuthority.name[selectedLanguage]}
+                    />
+                  ) : (
+                    <div className={classes.emptyLogo} />
+                  )}
+                  <div className={classes.serviceLabel}>{resource.title[selectedLanguage]}</div>
+                  <div>{resource.hasCompetentAuthority.name[selectedLanguage]}</div>
+                </div>
+              );
+            })}
+          </>
+        )}
+        {services?.length === 0 && (
+          <div className={classes.noServicesText}>
+            {t('policy_editor.access_package_no_services')}
+          </div>
+        )}
```

Comment:
There is a lot of different things going on in here. I suggest that we split this component up into smaller components to increase readability and to have one component to do one thing only 😄 
Maybe we can do something like this? Could potentially be split even more. 
Also, for texts we have StudioParagraph, StudioHeading and StudioLabelAsParagraph which I recommend to use 😄 


```tsx
        ...
        const hasServices: boolean = services?.length > 0;
        const serviceListIsEmpty: boolean = services?.length === 0;
        ...
        
        {hasServices && (
          <>
            <StudioParagraph size='xs' spacing className={classes.serviceContainerHeader}>
              {t('policy_editor.access_package_services')}
            </StudioParagraph>
            <Services services={services} />
          </>
        )}
        {serviceListIsEmpty && (
          <StudioParagraph size='xs' className={classes.noServicesText}>
            {t('policy_editor.access_package_no_services')}
          </StudioParagraph>
        )}

       ...

type ServicesProps = {
  services: AccessPackageResource[];
};

const Services = ({ services }: ServicesProps): ReactElement[] => {
  return services.map((resource: AccessPackageResource) => (
    <div key={resource.identifier} className={classes.serviceContainer}>
      <ResourceImage resource={resource} />
      <div className={classes.serviceLabel}>{resource.title[selectedLanguage]}</div>
      <div>{resource.hasCompetentAuthority.name[selectedLanguage]}</div>
    </div>
  ));
};

type ResourceImageProps = {
  resource: AccessPackageResource;
};
const ResourceImage = ({ resource }: ResourceImageProps): ReactElement => {
  if (resource.logoUrl) {
    return (
      <img
        className={classes.logo}
        src={resource.logoUrl}
        alt={resource.hasCompetentAuthority.name[selectedLanguage]}
        title={resource.hasCompetentAuthority.name[selectedLanguage]}
      />
    );
  }
  return <div className={classes.emptyLogo} />;
};
```